model/labels/regexp_test.go

```diff
@@ -1209,6 +1209,10 @@ func visitStringMatcher(matcher StringMatcher, callback func(matcher StringMatch
 func TestToNormalisedLower(t *testing.T) {
 	testCases := map[string]string{
 		"foo":                      "foo",
+		"FOO":                      "foo",
+		"Foo":                      "foo",
+		"foO":                      "foo",
+		"fOo":                      "foo",
```

Comment:
Would it be possible to incorporate a test case into both TestToNormalisedLower and BenchmarkToNormalizedLower? The test case should include all uppercase ASCII characters, with the final character being non-ASCII.
Something like "AAAAſ".

Reply:
I benchmarked this:

```
	b.Run("ascii uppercase then utf8", func(b *testing.B) {
		inputs := make([]string, 10)
		for i := range inputs {
			inputs[i] = benchCase(9, "all", true, i) + "Ж"
		}
		b.ResetTimer()
		for n := 0; n < b.N; n++ {
			toNormalisedLower(inputs[n%len(inputs)])
		}
	})
```
And the results were:

```
goos: darwin
goarch: arm64
pkg: github.com/prometheus/prometheus/model/labels
                                               │ main_asciithenunicode │     new_asciithenunicode     │
                                               │        sec/op         │   sec/op     vs base         │
ToNormalizedLower/ascii_uppercase_then_utf8-12             85.16n ± 0%   84.91n ± 1%  ~ (p=0.221 n=6)

                                               │ main_asciithenunicode │     new_asciithenunicode      │
                                               │         B/op          │    B/op     vs base           │
ToNormalizedLower/ascii_uppercase_then_utf8-12              32.00 ± 0%   32.00 ± 0%  ~ (p=1.000 n=6) ¹
¹ all samples are equal

                                               │ main_asciithenunicode │     new_asciithenunicode      │
                                               │       allocs/op       │ allocs/op   vs base           │
ToNormalizedLower/ascii_uppercase_then_utf8-12              2.000 ± 0%   2.000 ± 0%  ~ (p=1.000 n=6) ¹
¹ all samples are equal
```

Given that fitting this benchmark into current benchmarks structure would require a refactor, and it's not measuring any difference, I'd prefer to leave the benchmarks unmodified in this PR for the sake of clarity.